ocean_lib/ocean/ocean_exchange.py

```diff
@@ -98,8 +98,10 @@ def buy_at_fixed_rate(
                 raise VerifyTxFailed(
                     f"Approve OCEAN tokens failed, exchange address was {self._exchange_address} and tx id was {tx_id}!"
                 )
-        if ocean_token.balanceOf(wallet.address) < amount_base:
-            raise InsufficientBalance("Insufficient funds for buying DataTokens!")
+        if ocean_token.balanceOf(wallet.address) < ocean_amount_base:
```

Comment:
Can you please move this `if` block above the allowance check but after the validation check? I'd hate for someone to send an allowance transaction, only to realize afterward they don't have sufficient OCEAN tokens.